docs/development/commit-messages.md

```diff
@@ -0,0 +1,145 @@
+---
+layout: developer-doc
+title: Commit Messages
+category: development
+tags: [development, commit, git]
+order: 1
+---
+
+# Commit Messages
+
+Commit messages are an incredibly important part of development, as they provide
+a summary of the changes that are included in a commit. This is instrumental in
+producing a clear history for a given project.
+
+<!-- MarkdownTOC levels="2,3" autolink="true" -->
+
+- [Formatting Your Commit Messages](#formatting-your-commit-messages)
+  - [Summary Line](#summary-line)
+  - [Detailed Explanation](#detailed-explanation)
+  - [Issue Number](#issue-number)
+  - [General Formatting](#general-formatting)
+- [Where these Guidelines Apply](#where-these-guidelines-apply)
+
+<!-- /MarkdownTOC -->
+
+## Formatting Your Commit Messages
+
+Your commit messages should be formatted as follows:
+
+```plain
+[Type] A brief summary line
+
+1.  A numbered list.
+2.  That describes the changes made.
+
+Related #issue-no
+```
+
+All elements are mandatory for all commits. Please note that the blank line
+between the summary and the numbered list, and the line between the numbered
+list and the `Resolves` are crucial to prevent certain tools from getting
+confused.
+
+An example of a good commit message is as follows:
+
+```plain
+[Fix, Style] Fix a crash in the frobnicator
+
+1.  This commit fixes a bug in the method resolver that had arisen due
+    to an issue with the frobnicator. Tweaking the algorithm used by
+    the frobnicator from super-fancy-algo to even-fancier-algo resolved
+    the issue.
+2.  It also fixes a minor style issue with the naming of builder
+    methods for the Truffle nodes.
+
+Resolves #211
+Resolves #283
+```
+
+### Summary Line
+
+The summary line is a short descriptor of the commit. It must meet the following
+requirements:
+
+1.  It is written as an imperative "Create a new API for
+    $foo", not "Creates a new API for $foo".
+2.  It is less than 50 characters long.
```

Comment:
Is a summary line 50 characters long also accepted? (Is this a `<` or `<=`?)

Reply:
It should be `<=`. Will reword.